client/wfnews-war/src/main/angular/src/app/components/full-details/full-details.component.ts

```diff
@@ -48,8 +48,14 @@ document.getElementById('mobile-navigation-bar').style.display = 'block';
 
   back() {
     try {
-      if (this.params && this.params['source']) {
-        if (
+      if (this.params?.['source']) {
+        // use query params to determine the layer, coordinates and zoom level for routing back to the map
+        if ((this.params['source'] === 'map' || this.params['source']?.[0] === 'map')
+          && (this.params?.['type'] === 'area-restriction' || this.params?.['type'] === 'bans-prohibitions'
+            || this.params?.['type'].includes('evac'))
+        ) {
```

Comment:
How would we summarize this logic into plain english?

Reply:
If the user accessed the full details page from the map, and if the full details page contains one of area restrictions, fire bans, evac orders or evac alerts, use the backToMap function to route back to the map using the same layer and zoom level along with the appropriate coordinates for that type.
I updated the comments in the code with this explanation

Reply:
Logic updated to make it more readable